### PR TITLE
Re-organise 'Contributing' section

### LINKED ISSUE #281: Reorganise 'Contributing guide'
## 🧰 Task

Proposal to reorganise [Contributing guide](https://napari.org/dev/developers/contributing.html):

```
Contributing guide
|-- How to contribute                          # Reorganise of current contributing guide page
|   |-- Setting up a development installation   
|   |-- Opening a Pull Request  
|   |-- Adding tests
|   |-- Translations
|   |-- Adding an icon  
|   `-- Performance
|   |   |-- Benchmarks
|   |   `-- Profiling
|   `-- Contributing documentation 
`-- Napari architecture guide
    |-- Directory structure                   # To be added in new PR
    |-- An introduction to the event loop     # Migrate from 'In-depth explanations'
    |-- Napari models and events              # Migrate from 'In-depth explanations'
    `-- App-model                             # Migrate from 'In-depth explanations'

```

* 'How to contribute' would include an index
* The pages to migrate from 'In-depth explanations' are really aimed at devs, and state this at the start of the docs. For "An introduction to the event loop" only the section after 'in depth' will be migrated.

Related #25, this would be a small step towards improving napari architecture docs. The reorganisation would be in one PR and the "Directory structure" section would be a separate PR.

### PR DESCRIPTION
**Pull request by lucyleeow**
# References and relevant issues
closes #281

# Description

- Add a 'Contributing guide' section, with all relevant items (translations, testing etc ) nested here (see  #281 for details)
- Expand and link the performance monitoring and profiling parts
- Remove the `contributing.md` section, and move items to their own file or add to the new index page for the contributing guide section.

### CODE REVIEW

File: docs/developers/contributing/performance/profiling.md
Comment (lucyleeow) [inline docs/developers/contributing/performance/profiling.md]: I was trying to add more information about when perfmon may be useful, relative to profiling. Was from conversation with @jni but this is not fleshed out and I don't fully understand some points. Needs work. Suggestions welcome.

**Comment by lucyleeow** [comment]
cc @jni

File: docs/developers/contributing/performance/profiling.md
Comment (lucyleeow) [reply docs/developers/contributing/performance/profiling.md]: Okay updated after more discussion, but happy to make any changes.

**Comment by lucyleeow** [comment]
ping @melissawm or @psobolewskiPhD may be interested in reviewing? :pray:

**Comment by psobolewskiPhD** [comment]
Yes, sorry it's on my list. Hopefully later today or tomorrow.

File: docs/developers/index.md
Comment (psobolewskiPhD) [inline docs/developers/index.md]:
```suggestion
    - [Testing](napari-testing)
    - [Translations](translations)
```
matching order in ToC

File: docs/_toc.yml
Comment (psobolewskiPhD) [inline docs/_toc.yml]:
```suggestion
              - file: developers/contributing/performance/profiling
              - file: developers/contributing/performance/benchmarks
```
Nothing shows up in the subtree in the sidebar for Performance:
image

File: docs/developers/index.md
Comment (psobolewskiPhD) [inline docs/developers/index.md]:
```suggestion
  on the Image.sc forum. All usage support requests should be posted on the forum with
```

File: docs/developers/contributing/performance/index.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/performance/index.md:6]: This is just showing a blank area
image

But this could be due to the issue in the _toc.yml

File: docs/developers/contributing/performance/profiling.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/performance/profiling.md]:
```suggestion
![A section of an example call graph showing the functions called in napari's layerlist and some of their profile statistics such as call count and cumulative percentage time spent.](../images/execution_graph.png)
```
Or move the image to /performance

File: docs/howtos/perfmon.md
Comment (psobolewskiPhD) [inline docs/howtos/perfmon.md]:
```suggestion
[our profiling documentation](profiling).
```

**Review by psobolewskiPhD — CHANGES_REQUESTED** [review]
Love it!
Made some small suggestions. also:
- testing.md was moved, but not edited. It appears to have a relative link to an image, because the image doesn't load

While we're doing this, what do you think of making a ToC subtree for the extra core dev stuff that's all alone now (see below):
image
Not sure what I'd name it, maybe just `Core Developer Guide` like it is and nest those underneath?

File: docs/_toc.yml
Comment (lucyleeow) [reply docs/_toc.yml]: Fixed!

**Comment by lucyleeow** [comment]
@psobolewskiPhD thank you for reviewing. Changes made.

The toc in the `docs/developers/coredev/index.md` page is a bit long due to many headings in the pages. I tried adding `:depth:` and `:maxdepth:` but neither worked. Maybe it is fine as is?

File: docs/developers/index.md
Comment (psobolewskiPhD) [inline docs/developers/index.md]:
```suggestion
  - [Core Developer guide](core-developer-guides)
```

File: docs/developers/index.md
Comment (psobolewskiPhD) [inline docs/developers/index.md]:
```suggestion
  - [Maintenance](coredev/maintenance)
```
Or, better yet add a reference in maintenance.md?

File: docs/developers/index.md
Comment (psobolewskiPhD) [inline docs/developers/index.md]:
```suggestion
  - [Release guide](coredev/release)
```
Same as above

**Comment by psobolewskiPhD** [comment]
I caught a few nonfunctional links.
Also coredev/maintenance.md has broken images because of relative links not resolving. Missed this before, sorry!
Plus it looks like we got some conflicts from the PR with references? 😬

**Comment by lucyleeow** [comment]
Yesss I thought there would be a huge merge problem! Oh dear.

File: docs/developers/index.md
Comment (lucyleeow) [reply docs/developers/index.md]: There's already a reference!

**Comment by lucyleeow** [comment]
@psobolewskiPhD merge conflicts fixed and the rendered docs look okay...

File: docs/developers/contributing/performance/profiling.md
Comment (psobolewskiPhD) [reply docs/developers/contributing/performance/profiling.md]: Doh I think I got this wrong. It's like double deep I think.

File: docs/developers/contributing/performance/profiling.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/performance/profiling.md]:
```suggestion
![A section of an example call graph showing the functions called in napari's layerlist and some of their profile statistics such as call count and cumulative percentage time spent.](../../images/execution_graph.png)
```
The images are in /developers/images
and this file is in:
/developers/contributing/performance
So I think I have it correct now?

File: docs/developers/contributing/performance/profiling.md
Comment (psobolewskiPhD) [reply docs/developers/contributing/performance/profiling.md]: I made a new suggestion.

File: docs/developers/contributing/performance/profiling.md
Comment (psobolewskiPhD) [reply docs/developers/contributing/performance/profiling.md]: It's that or we move the images folder.

File: docs/developers/contributing/index.md
Comment (melissawm) [inline docs/developers/contributing/index.md]:
```suggestion
Note we are pushing to the remote named 'origin' above. If you have followed the
```

**Review by melissawm — APPROVED** [review]
This looks great to me, thanks @lucyleeow !

**Comment by lucyleeow** [comment]
Thanks @melissawm @psobolewskiPhD ! Changes made!

**Comment by psobolewskiPhD** [comment]
Thanks! the image works!

I swear I'm not trying to be a PITA, especially if I'm the one that requested this in the first place, but:
Clicking in the Sidebar ToC for Contributing on the left on `Core dev guide` brings you to a page with another ToC:
https://output.circle-artifacts.com/output/job/397a1f5e-9b42-4b09-a183-44646dc8402e/artifacts/0/docs/docs/_build/developers/coredev/index.html
rather than the actual guide:
https://output.circle-artifacts.com/output/job/397a1f5e-9b42-4b09-a183-44646dc8402e/artifacts/0/docs/docs/_build/developers/coredev/core_dev_guide.html

None of the other sidebar ToC items behave that way in Contributing?

On the other hand, the Release notes landing page is basically empty, maybe it should just list all the release notes ToC style?

**Comment by lucyleeow** [comment]
Yes so, when you click on a header on the sidebar (e.g., contributing guide, core-dev guide), it takes you to the `index.md` page for that folder (e.g., for contributing guide, it takes you to `docs/developers/contributing/index.md`). For core-dev-guide though, I didn't have anything to add so I just added a toc (which is annoyingly too detailed, see: https://github.com/napari/docs/pull/318#issuecomment-1903307110).

But I agree, I can amend that index page to just a one liner saying e.g., 'core dev guides...' and I can manually add a TOC with just the pages (i.e., max depth 1) instead of the full TOC. WDYT? Did that make sense?

**Comment by psobolewskiPhD** [comment]
Could we just rename `core_dev_guide.md` to `index.md` ?

**Comment by lucyleeow** [comment]
I guess so but then it doesn't match the organisation of the other ones. I think the index should be an intro and it should at least mention all the other pages within that header? (i.e., introduce the whole section and what pages are in that section)

**Comment by psobolewskiPhD** [comment]
Ok, I see.
What if index.md had:
https://output.circle-artifacts.com/output/job/397a1f5e-9b42-4b09-a183-44646dc8402e/artifacts/0/docs/docs/_build/developers/coredev/core_dev_guide.html#core-developer-guide
and 
https://output.circle-artifacts.com/output/job/397a1f5e-9b42-4b09-a183-44646dc8402e/artifacts/0/docs/docs/_build/developers/coredev/core_dev_guide.html#all-contributors-are-treated-the-same

And then you'd just navigate to the next sections using the bottom `>` or the toc?

Trying to mimic the general contributing guide landing page.

**Comment by melissawm** [comment]
I don't think we actually need to rename the file to index - the organization should obey the _toc.yml file so if we had
```
      - file: developers/contributing/core_dev_guide
        subtrees:
        - entries:
          - file: developers/contributing/dev_install
          - file: developers/contributing/testing
          - file: developers/contributing/translations
          - file: developers/contributing/performance/index
            subtrees:
            - entries:
              - file: developers/contributing/performance/profiling
              - file: developers/contributing/performance/benchmarks
          - file: developers/contributing/documentation/index
            subtrees:
            - entries:
              - file: developers/contributing/documentation/docs_template
              - file: developers/contributing/documentation/docs_deployment
      - file: developers/coredev/index
        subtrees:
        - entries:
          - file: developers/coredev/core_dev_guide
          - file: developers/coredev/maintenance
          - file: developers/coredev/release
          - file: developers/coredev/packaging
 ```
I guess this would be enough? 

Another option is to do like we did for other subpages and use cards instead of plain tocs - but this could be a follow-up.

**Comment by lucyleeow** [comment]
Oh yeah that's a good point! Also what is a 'card'?

I do think the page should introduce the section and the pages in the section. I'll have a look at that page and see if it mentions the other pages (release, maintenance etc)

**Comment by psobolewskiPhD** [comment]
ooo! I love cards!
Cards are those neato things like here:
https://output.circle-artifacts.com/output/job/397a1f5e-9b42-4b09-a183-44646dc8402e/artifacts/0/docs/docs/_build/usage.html

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by psobolewskiPhD** [comment]
I'm going to approve and mark ready-to-merge and if you come up with something cool, great. if not, we'll merge and make an issue to improve?

**Comment by lucyleeow** [comment]
Okay so reading 'Core Developer guide', the 'Further resources' does point to lots of things and I will update to add the other pages in the section. It looks like a good landing/index page and I will update the toc as @melissawm suggested.

Cards look great, but this improvement could be added later as I can't see an easy way to add cards to the 'Core Developer guide' section.

**Comment by psobolewskiPhD** [comment]
Looks great now!
Thank y'all for your efforts! 
I will try to remember to merge tomorrow, but if I don't please @ me a reminder!
❤️

**Comment by lucyleeow** [comment]
No worries, thanks. I was not sure of the wording here though:

```md
### Project management guides

- Our [code of conduct](napari-coc).
- Our [governance](napari-governance).
- Our [mission and values](mission-and-values).
```

Is 'Project management' the right word...?

**Comment by melissawm** [comment]
I'd say it is appropriate, but if you want to explore options how about "Project organization" or just "Community" as it is right now? Edit: what I mean is this toc is exactly what is under the Community tab in the home page

**Comment by lucyleeow** [comment]
Thanks, I think 'community' sounds nicer. @psobolewskiPhD this is ready to go in if you're happy! Thank you!

**Pull request merged**
